sklearn/preprocessing/_discretization.py

```diff
@@ -267,7 +278,28 @@ def fit(self, X, y=None):
                 bin_edges[jj] = np.linspace(col_min, col_max, n_bins[jj] + 1)
 
             elif self.strategy == "quantile":
-                quantiles = np.linspace(0, 100, n_bins[jj] + 1)
+                # if sample_weight (weight) parameter is defined, then the size
```

Comment:
we should raise an error if `sample_weight` is passed with the other strategies to not ignore it silently.

Reply:
I corrrected this in the following commit : https://github.com/scikit-learn/scikit-learn/pull/22048/commits/85fe2b74ee5fe571b92f113f9dd8182339a4c4aa

I now raise an error when the `sample_weight` parameter is used with the wrong strategy on line 237 in _discretization.py